nixos/modules/programs/steam.nix

```diff
@@ -141,6 +143,28 @@ in {
       Load the extest library into Steam, to translate X11 input events to
       uinput events (e.g. for using Steam Input on Wayland)
     '';
+
+    protontricks = mkOption {
+      description = "Protontricks Steam integration";
+      default = {};
+      type = types.submodule {
+        options = {
+          enable = mkEnableOption "Protontricks";
+          package = mkOption {
+            type = types.package;
+            default = pkgs.protontricks;
+            defaultText = literalExpression "pkgs.protontricks";
+            apply = protontricks: protontricks.overrideAttrs (prev: {
+              makeWrapperArgs = prev.makeWrapperArgs ++ (lib.optional (cfg.extraCompatPackages != [ ])
+                "--set STEAM_EXTRA_COMPAT_TOOLS_PATHS ${extraCompatPath}");
+            });
```

Comment:
Move this to the implementation. `apply` is for when you to do some change so that it's visible to Nix code reading the same config option.

Reply:
I'm not sure I understand, could you maybe give me an example?

Reply:
Then again, what's the reason it's used for the steam package?

Reply:
For it I could understand it slightly more as I could imagine usage by users outside of the steam module but it too should probably be moved to the implementation.

Note that this is just a minor issue.

Reply:
I see, so it's only if you want to expose it for external usage.

That's all my questions then, I'll try to implement the suggestions tomorrow.